src/main/java/de/tum/in/www1/artemis/domain/quiz/QuizConfiguration.java

```diff
@@ -0,0 +1,109 @@
+                    }
+                }
+                if (quizQuestion instanceof DragAndDropQuestion dragAndDropQuestion) {
+                    DragAndDropQuestionStatistic dragAndDropStatistic = (DragAndDropQuestionStatistic) dragAndDropQuestion.getQuizQuestionStatistic();
+                    // reconnect dropLocations
+                    for (DropLocation dropLocation : dragAndDropQuestion.getDropLocations()) {
+                        if (dropLocation.getId() != null) {
+                            dropLocation.setQuestion(dragAndDropQuestion);
+                        }
+                    }
+                    // reconnect dragItems
+                    for (DragItem dragItem : dragAndDropQuestion.getDragItems()) {
+                        if (dragItem.getId() != null) {
+                            dragItem.setQuestion(dragAndDropQuestion);
+                        }
+                    }
+                    // reconnect correctMappings
+                    for (DragAndDropMapping mapping : dragAndDropQuestion.getCorrectMappings()) {
+                        if (mapping.getId() != null) {
+                            mapping.setQuestion(dragAndDropQuestion);
+                        }
+                    }
+                    // reconnect dropLocationCounters
+                    for (DropLocationCounter dropLocationCounter : dragAndDropStatistic.getDropLocationCounters()) {
+                        if (dropLocationCounter.getId() != null) {
+                            dropLocationCounter.setDragAndDropQuestionStatistic(dragAndDropStatistic);
+                            dropLocationCounter.getDropLocation().setQuestion(dragAndDropQuestion);
+                        }
+                    }
+                }
+                if (quizQuestion instanceof ShortAnswerQuestion shortAnswerQuestion) {
+                    ShortAnswerQuestionStatistic shortAnswerStatistic = (ShortAnswerQuestionStatistic) shortAnswerQuestion.getQuizQuestionStatistic();
+                    // reconnect spots
+                    for (ShortAnswerSpot spot : shortAnswerQuestion.getSpots()) {
+                        if (spot.getId() != null) {
+                            spot.setQuestion(shortAnswerQuestion);
+                        }
+                    }
+                    // reconnect solutions
+                    for (ShortAnswerSolution solution : shortAnswerQuestion.getSolutions()) {
+                        if (solution.getId() != null) {
+                            solution.setQuestion(shortAnswerQuestion);
+                        }
+                    }
+                    // reconnect correctMappings
+                    for (ShortAnswerMapping mapping : shortAnswerQuestion.getCorrectMappings()) {
+                        if (mapping.getId() != null) {
+                            mapping.setQuestion(shortAnswerQuestion);
+                        }
+                    }
+                    // reconnect spotCounters
```

Comment:
This is the same for loop again. Also generalize this please.

Reply:
done